src/CommandingOfficers/RuinedCommander.java

```diff
@@ -0,0 +1,436 @@
+package CommandingOfficers;
+
+import Engine.Army;
+import Engine.GameAction;
+import Engine.GameActionSet;
+import Engine.GameInstance;
+import Engine.GamePath;
+import Engine.GameScenario;
+import Engine.UnitActionFactory;
+import Engine.Utils;
+import Engine.XYCoord;
+
+import java.awt.Color;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+
+import CommandingOfficers.CommanderInfo.InfoPage;
+import Engine.Combat.StrikeParams;
+import Engine.Combat.StrikeParams.BattleParams;
+import Engine.GameEvents.CommanderAbilityEvent;
+import Engine.GameEvents.GameEventQueue;
+import Engine.StateTrackers.KillCountsTracker;
+import Engine.StateTrackers.StateTracker;
+import Engine.UnitMods.UnitModifierWithDefaults;
+import Terrain.GameMap;
+import Terrain.MapMaster;
+import UI.GameOverlay;
+import Units.Unit;
+import Units.UnitContext;
+import Units.UnitDelta;
+
+public abstract class RuinedCommander extends DeployableCommander
+{
+  private static final long serialVersionUID = 1L;
+  public static final InfoPage DOR_MECHANICS_BLURB = new InfoPage(
+            "Days of Ruin mechanics:\n"
+          + "You have a CO zone that covers the area around your COU, and grants bonuses to your units inside.\n"
+          + "The zone expands 1 tile when your meter is half full, and again at full charge.\n"
+          + "All units get +10 attack and +10 Damage Division (DD) while in the zone.\n"
```

Comment:
Is "Damage Division" like, a real term?

Reply:
It is now.

Reply:
You gotta define your terms, man. You always just invent them and then expect people to understand what you mean. The obvious meaning is that incoming damage is now divided by 10, but that can't be right.

Reply:
It's Advance Wars, so everything has a base of 100.
You're right, I should write up that readme, though.
Should probably move most of this to that, too? It's uh... kind of a lot for the CO page.

Reply:
So "Damage Division" is just "Defense, but calculated according to Days of Ruin rules". Yeah, it's confusing that it's calculated differently in the different games, and more confusing that we are combining them both in one game. It's probably good to use a separate term for clarity, though I might even just spell it out since we don't have a glossary, e.g.: units get +10% attack and take 10% less damage while in the zone." It really isn't more verbose, and I don't think I'd stress about repeating it on all of them.

Reply:
The problem with that phrasing is that it applies equally well to either type of defense.
You will note that I generally leave of percents for attack as well, since +10% attack could just as easily be a 1.1x multiplier to your damage as a +10 to your attack value.

Trilogy defense makes you take X% less of the opponent's entire modified damage output, regardless of your previous stat (there are AW1-shaped asterisks there, but bleeeh)
DoR defense makes you take X% of your existing defense stat less of the opponent's already-modified damage output. (that's not quite an exact description, I think but you get the gist)

Reply:
Admittedly the language is overloaded, which creates confusion. This is why it's important to clarify. Maybe even just "DoR Defense" might be clear enough since the nerds will know what that means and the normies won't care. Making up new terms could be helpful, but only if they have discoverable definitions (and I have little hope that anyone is gonna bust out the README - I sure wouldn't think to look there). Might not even hurt to rename _both_ quantities just to create distance from the overloaded term.

Also, for your amusement, and because LLMs are fun:
https://www.perplexity.ai/search/f0e12951-2024-4559-abce-7de865255750?s=u

Reply:
Yeah. I was thinking I ought to write a README on the damage calculations, but it doesn't really matter if nobody reads it.
Jay said it sounded like a good idea, but my level of confidence in that solution is not high.

That is amusing, but those proposed names aren't super clearly related to either defense model either.

Reply:
Yeah, and yeah. A game glossary kinda has to be in-game to be useful, especially nowadays.

And yes; this is a hard naming problem, since they are very related calculations. It might not hurt to just call it DoR defense instead of trying to come up with fancy names.

Reply:
Yeah, fair enough.